src/Workspaces/Remote/ServiceHub/Services/NavigateToSearch/RemoteNavigateToSearchService.cs

```diff
@@ -55,13 +67,14 @@ public ValueTask HydrateAsync(Checksum solutionChecksum, CancellationToken cance
             RemoteServiceCallbackId callbackId,
             CancellationToken cancellationToken)
         {
-            return RunServiceAsync(solutionChecksum, async solution =>
+            return RunServiceAsync(solutionChecksum, solution =>
             {
                 var document = solution.GetRequiredDocument(documentId);
-                var callback = GetCallback(callbackId, cancellationToken);
-
-                await AbstractNavigateToSearchService.SearchDocumentInCurrentProcessAsync(
-                    document, searchPattern, kinds.ToImmutableHashSet(), callback, cancellationToken).ConfigureAwait(false);
+                return PushToCallbackAsync(
+                    callbackId,
+                    AbstractNavigateToSearchService.SearchDocumentInCurrentProcessAsync(
+                        document, searchPattern, kinds.ToImmutableHashSet(), cancellationToken),
+                    cancellationToken);
             }, cancellationToken);
```

Comment:
on the oop side, we call into teh services normally, getting back an IAE for them.  Then in PushToCallbackAsync, we simply take those results and serialize them into the passed in callback.